### PR TITLE
3350 drops pytorch 1.5.x support

### LINKED ISSUE #3350: nightly build:  test_milmodel
**Describe the bug**
```
======================================================================
ERROR: test_script (tests.test_milmodel.TestMilModel)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins/agent/workspace/Monai-pytorch-versions/tests/test_milmodel.py", line 87, in test_script
    test_script_save(net, test_data)
  File "/home/jenkins/agent/workspace/Monai-pytorch-versions/tests/utils.py", line 597, in test_script_save
    atol=atol,
  File "/home/jenkins/agent/workspace/Monai-pytorch-versions/monai/networks/utils.py", line 465, in convert_to_torchscript
    script_module = torch.jit.script(model)
  File "/opt/conda/lib/python3.6/site-packages/torch/jit/__init__.py", line 1261, in script
    return torch.jit._recursive.create_script_module(obj, torch.jit._recursive.infer_methods_to_compile)
  File "/opt/conda/lib/python3.6/site-packages/torch/jit/_recursive.py", line 305, in create_script_module
    return create_script_module_impl(nn_module, concrete_type, stubs_fn)
  File "/opt/conda/lib/python3.6/site-packages/torch/jit/_recursive.py", line 361, in create_script_module_impl
    create_methods_from_stubs(concrete_type, stubs)
  File "/opt/conda/lib/python3.6/site-packages/torch/jit/_recursive.py", line 279, in create_methods_from_stubs
    concrete_type._create_methods(defs, rcbs, defaults)
  File "/opt/conda/lib/python3.6/site-packages/torch/jit/_recursive.py", line 587, in compile_unbound_method
    create_methods_from_stubs(concrete_type, (stub,))
  File "/opt/conda/lib/python3.6/site-packages/torch/jit/_recursive.py", line 279, in create_methods_from_stubs
    concrete_type._create_methods(defs, rcbs, defaults)
RuntimeError: 
cannot call a value of type 'None':
  File "/home/jenkins/agent/workspace/Monai-pytorch-versions/monai/networks/nets/milmodel.py", line 182

            x = x.permute(1, 0, 2)
            x = self.transformer(x)
                ~~~~~~~~~~~~~~~~ <--- HERE
            x = x.permute(1, 0, 2)
'MILModel.calc_head' is being compiled since it was called from 'MILModel.forward'
  File "/home/jenkins/agent/workspace/Monai-pytorch-versions/monai/networks/nets/milmodel.py", line 227

        if not no_head:
            x = self.calc_head(x)
            ~~~~~~~~~~~~~~~~~~~~ <--- HERE

        return x


----------------------------------------------------------------------
```

```
================================
Printing MONAI config...
================================
MONAI version: 0.7.0+142.gb7daf804
Numpy version: 1.18.1
Pytorch version: 1.5.1
MONAI flags: HAS_EXT = True, USE_COMPILED = True
MONAI rev id: b7daf804110081f4e80ea0c91604f99279f4116b

Optional dependencies:
Pytorch Ignite version: 0.4.6
Nibabel version: 3.2.1
scikit-image version: 0.15.0
Pillow version: 8.4.0
Tensorboard version: 2.1.0
gdown version: 4.2.0
TorchVision version: 0.6.1
tqdm version: 4.62.3
lmdb version: 0.98
psutil version: 5.6.7
pandas version: 0.24.2
einops version: 0.3.2
transformers version: 4.12.4
mlflow version: 1.21.0

For details about installing the optional dependencies, please visit:
    https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies


================================
Printing system config...
================================
System: Linux
Linux version: Ubuntu 18.04.4 LTS
Platform: Linux-5.4.0-65-generic-x86_64-with-debian-buster-sid
Processor: x86_64
Machine: x86_64
Python version: 3.6.9
Process name: python
```

### PR DESCRIPTION
**Pull request by wyli**
Signed-off-by: Wenqi Li <wenqil@nvidia.com>

part of #3350

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Review by Nic-Ma — COMMENTED** [review]
Hi @wyli ,

Could you please help also drop this kind of lines:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/utils.py#L1313

Thanks.

File: .github/workflows/pythonapp-gpu.yml [hunk 1]
```diff
@@ -19,17 +19,13 @@ jobs:
     strategy:
       matrix:
         environment:
-          - "PT16+CUDA110"
```
[the remark this answers is not in the document]
Comment (wyli) [reply .github/workflows/pythonapp-gpu.yml:22]: Hi @Nic-Ma, this docker image is based on nvidia 1.6.0aXXX, it is before the pytorch official 1.6, doesn't satisfy "torch>=1.6"

[the remark this answers is not in the document]
Comment (Nic-Ma) [reply .github/workflows/pythonapp-gpu.yml:22]: Oh, I see!
Thanks.

**Review by Nic-Ma — APPROVED** [review]

**Comment by wyli** [comment]
> Hi @wyli ,
> 
> Could you please help also drop this kind of lines: https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/utils.py#L1313 If you don't plan to change these torch APIs in this PR, I can help do it soon.
> 
> Thanks.

thanks, updated the PR to remove `hasattr(torch, "searchsorted")`, the other checks look necessary. once this is merged we can check the code coverage report to remove the necessary ones.

**Comment by Nic-Ma** [comment]
> > Hi @wyli ,
> > Could you please help also drop this kind of lines: https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/utils.py#L1313 If you don't plan to change these torch APIs in this PR, I can help do it soon.
> > Thanks.
> 
> thanks, updated the PR to remove `hasattr(torch, "searchsorted")`, the other checks look necessary. once this is merged we can check the code coverage report to remove the necessary ones.

Sure, sounds good to me.
I already manually checked several `hasattr(torch, "XXX")`  in PyTorch 1.5.1, not much need to update I think.

Thanks.

**Pull request merged**